app/components/primer/alpha/segmented_control.ts

```diff
@@ -0,0 +1,38 @@
+import {controller, targets} from '@github/catalyst'
+
+@controller
+export class SegmentedControlElement extends HTMLElement {
+  @targets buttons: HTMLElement[]
+
+  connectedCallback() {
+    this.#updateButtonLabels()
+  }
+
+  select(event: Event) {
+    const button = event.currentTarget as HTMLButtonElement
+    for (const item of this.querySelectorAll('li.SegmentedControl-item')) {
```

Comment:
Why not use the `buttons` target defined above instead of using `querySelector`?